dolphin/drivers/dellemc_storage/vmax.py

```diff
@@ -0,0 +1,223 @@
+            LOG.error("Invalid input access_info[" + str(err) + "]")
+            return None
+
+        try:
+            # Get optional inputs
+            if 'extra_attributes' in access_info:
+                extra_attr = access_info['extra_attributes']
+
+                if 'verify' in extra_attr:
+                    self.verify = extra_attr['verify']
+                if 'univmax_ver' in extra_attr:
+                    self.univmax_version = extra_attr['univmax_ver']
+                if 'array_id' in extra_attr:
+                    self.symmetrix_id = extra_attr['array_id']
+
+        except KeyError as err:
+            LOG.info("Invalid input access_info[" + str(err) + "]")
+
+        description = 'Unispere array with version '
+        try:
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                u4v_version=self.univmax_version,
+                server_ip=self.server_ip,
+                port=self.port,
+                verify=False,
+                username=self.username,
+                password=self.password)
+
+            # Get the Unisphere version
+            version = conn.common.get_uni_version()
+            description = description + version[0]
+
+            if self.symmetrix_id == "":
+                # Retrieve a list of arrays managed by Unisphere
+                array_list = conn.common.get_array_list()
+
+                if len(array_list) == 0:
+                    LOG.error("Array list from Storage backend is empty")
+                    conn.close_session()
+                    return None
+
+                if len(array_list) > 1:
+                    LOG.info("More than one Storage Array, first one selected")
+
+                self.symmetrix_id = array_list[0]
+
+            # Close the session
+            conn.close_session()
+
+        except:
+            LOG.error("Failed to connect to VMAX storage")
+            return None
+
+        # Get storage details
+        storage = self.get_storage(context)
+        if storage:
+            storage['description'] = description
+            storage['serial_number'] = self.symmetrix_id
+
+        # Return all the storage ids this driver manage
+        LOG.info("register_storage(), driver successfully registered")
+        return storage
+
+    def get_storage(self, context):
+
+        status = "Error"
+        model_symmetrix = ""
+        total_cap = 0
+        used_cap = 0
+
+        try:
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                u4v_version=self.univmax_version,
+                server_ip=self.server_ip,
+                port=self.port,
+                verify=False,
+                username=self.username,
+                password=self.password)
+
+            LOG.info("Successfully connected storage")
+            status = "Available"
+
+            # Get the Unisphere model
+            uri = "/system/symmetrix/" + self.symmetrix_id
```

Comment:
you can move some unishpere rest methods into another file

Reply:
We can handle this in another PR